webapp/__init__.py

```diff
@@ -1,40 +1,78 @@
-import os
-from flask import Flask, render_template
-
-
-def create_app(test_config=None):
-    # create and configure the app
-    app = Flask(__name__, instance_relative_config=True)
-
-    # Todo: SECRET_KEY should be overridden with a random value when deploying, e.g. a long random string of bytes
-    app.config.from_mapping(
-        # for signing session cookies or any other security related needs by extensions or your application
-        # set to 'dev' for development convenience
-        SECRET_KEY='dev',
-        # Path to database file
-        DATABASE=os.path.join(app.instance_path, 'webapp.sqlite'),
-    )
-
-    if test_config is None:
-        # load the instance config, if it exists, when not testing
-        app.config.from_pyfile('config.py', silent=True)
-    else:
-        # load the test config if passed in
-        app.config.from_mapping(test_config)
-
-    # ensure the instance folder exists
-    try:
-        os.makedirs(app.instance_path)
-    except OSError:
-        pass
+from flask import Flask
+from flask import render_template
+from flask import request
+from firebase_admin import auth
+from . import db
+from functools import wraps
+
+
+# init the app instance
+app = Flask(__name__)
+
+# init connection to Firebase
+# pb is used for db operations
+# firebase is an admin sdk for validating tokens
+firebase, pb = db.init_db()
 
-    # a simple page that says hello
-    @app.route('/hello')
-    def hello():
-        return 'Hello, World!'
 
-    return app
+# middleware to make sure only valid users can perform CRUD
+def check_token(f):
+    @wraps(f)
+    def wrap(*args,**kwargs):
+        if not request.headers.get('authorization'):
+            return {'message': 'No token provided'}, 400
+        try:
+            user = auth.verify_id_token(request.headers['authorization'])
+            request.user = user
+        except:
+            return {'message':'Invalid token provided.'}, 400
+        return f(*args, **kwargs)
+    return wrap
+
+
+# a hello page for testing
+@app.route('/hello')
```

Comment:
You may want to split up the type of requests being handled here.
Also, it's best to handle different APIs in different files.

Reply:
refactored!